core/src/main/java/org/apache/brooklyn/feed/http/HttpFeed.java

```diff
@@ -197,6 +197,10 @@ public Builder credentials(String username, String password) {
             this.credentials = new UsernamePasswordCredentials(username, password);
             return this;
         }
+        public Builder credentials(Credentials credentials) {
```

Comment:
Not sure about this. Nothing in the HttpFeed's public API talks about `org.apache.http.auth`. If we add this public method, then `org.apache.http.auth.Credentials` is exposed (rather than it being an implementation detail).

Mind you, some of our other "public" classes (e.g. `HttpTool`) exposes `org.apache.http.*` in its API.

What is the use-case for adding this?

@rdowner what are your thoughts - should we expose this in the API?